protocols/secio/src/stream_cipher.rs

```diff
@@ -33,20 +34,33 @@ impl Cipher {
         match *self {
             Cipher::Aes128 => 16,
             Cipher::Aes256 => 32,
+            Cipher::Null => 0
         }
     }
 
     /// Returns the size of in bytes of the IV expected by the cipher.
     #[inline]
     pub fn iv_size(&self) -> usize {
-        16      // CTR 128
+        match self {
+            Cipher::Aes128 | Cipher::Aes256 => 16,
+            Cipher::Null => 0
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct NullCipher;
+
+impl SynchronousStreamCipher for NullCipher {
+    fn process(&mut self, input: &[u8], output: &mut [u8]) {
+        output.copy_from_slice(input)
```

Comment:
Maybe a stupid question, but isn't there supposed to be an encoding side and a decoding side?

Reply:
AFAIK stream ciphers usually work by `xor`-ing a stream of pseudo-random numbers and the message. The symmetric key acts like a seed value for the random number generation. `xor`-ing the same sequence of pseudo-random numbers and the ciphertext gives back the original message. Therefore one `process` method fully specifies the workings of the cipher.